src/providers/workers-ai/chatComplete.ts

```diff
@@ -0,0 +1,189 @@
+                  item.image_url &&
+                  item.image_url.url
+                ) {
+                  const parts = item.image_url.url.split(';');
+                  if (parts.length === 2) {
+                    const base64ImageParts = parts[1].split(',');
+                    const base64Image = base64ImageParts[1];
+                    const mediaTypeParts = parts[0].split(':');
+                    if (mediaTypeParts.length === 2 && base64Image) {
+                      const mediaType = mediaTypeParts[1];
+                      transformedMessage.content.push({
+                        type: 'image',
+                        source: {
+                          type: 'base64',
+                          media_type: mediaType,
+                          data: base64Image,
+                        },
+                      });
+                    }
+                  }
+                }
+              });
+              messages.push(transformedMessage as Message);
+            } else {
+              messages.push({
+                role: msg.role,
+                content: msg.content,
+              });
+            }
+          });
+        }
+
+        return messages;
+      },
+    },
+  ],
+  stream: {
+    param: 'stream',
+    default: false,
+  },
+};
+
+export interface WorkersAiErrorObject {
+  type: string;
+  message: string;
+}
+
+interface WorkersAiErrorResponse {
+  type: string;
+  error: WorkersAiErrorObject;
+}
+
+interface WorkersAiChatCompleteResponse {
+  result: {
+    response: string;
+  };
+  success: boolean;
+  errors: string[];
+  messages: string[];
+}
+
+interface WorkersAiChatCompleteStreamResponse {
+  response: string;
+  p?: string;
+}
+
+// TODO: cloudflare do not return the usage
+// TODO: return the model
+export const WorkersAiChatCompleteResponseTransform: (
+  response: WorkersAiChatCompleteResponse | WorkersAiErrorResponse,
+  responseStatus: number
+) => ChatCompletionResponse | ErrorResponse = (response, responseStatus) => {
+  if (responseStatus !== 200 && 'errors' in response) {
+    return {
+      error: {
+        message: response.errors?.join(','),
+        type: null,
+        param: null,
+        code: null,
+      },
+      provider: WORKERS_AI,
+    } as ErrorResponse;
+  }
+
+  if ('result' in response) {
+    // const { input_tokens = 0, output_tokens = 0 } = {};
+
+    return {
+      id: Date.now().toString(),
+      object: 'chat_completion',
+      created: Math.floor(Date.now() / 1000),
+      model: '',
+      provider: WORKERS_AI,
+      choices: [
+        {
+          message: { role: 'assistant', content: response.result.response },
+          index: 0,
+          logprobs: null,
+          finish_reason: '',
+        },
+      ],
+      // usage: {
+      //   prompt_tokens: input_tokens,
+      //   completion_tokens: output_tokens,
+      //   total_tokens: input_tokens + output_tokens,
+      // },
+    };
+  }
+
+  return {
```

Comment:
You can use the newly added error generator function instead of writing the whole object. (for this and complete as well)

Example:
```
  return generateInvalidProviderResponseError(response, WORKERS_AI);
```

You can also check other providers to get a reference of the latest refactoring that happened for provider functions.